styx-common/src/main/java/com/spotify/styx/serialization/Json.java

```diff
@@ -65,6 +65,10 @@ public static ByteString serialize(Object value) throws JsonProcessingException
     return ByteString.of(OBJECT_MAPPER.writeValueAsBytes(value));
   }
 
+  public static <T> T deserialize(ByteString json, Class<T> clazz) throws IOException {
```

Comment:
Should we remove the two following methods `deserializeEvent` and `deserializeTrigger` to use this generic one only?